migrations/blocked_tasks.js

```diff
@@ -0,0 +1,83 @@
+// constants
+finishedStatuses = ["success","failed"]
+
+function taskIsBlocked(task) {
+    for(dep of task.depends_on) {
+        if (dep.unattainable) {
+            return true
+        }
+    }
+
+    return false
+}
+
+function unattainableResolved(task) {
+    for (dep of task.depends_on) {
+        if (dep.unattainable == null) {
+            return false
+        }
+    }
+    return true
+}
+
+//
+// example invocation:
+//
+// startingDate = ISODate("2016-01-01T00:00:00Z")
+// millisecondsPerDay = 86400000
+// delta = millisecondsPerDay * 1
+// sleepMilliseconds = 100
+// migrate(startingDate, delta, sleepMilliseconds)
+
+function migrate(startingDate, delta, sleepMilliseconds) {
```

Comment:
don't we want to migrate everything, and woudln't it be better to a limit, rather than a time window? We still have to run the filter on every document in this window... the window approach makes it easier to make the limit more reproducible (which is a good thing.)

Reply:
my thinking in not using a limit was to guarantee all the tasks in a version are being processed in one iteration of the outer while loop. Otherwise, it's possible we would spin on dependencies that aren't resolved. 

For example, if _A_ and _B_ depend on _C_ and _C_ depends on _D_ and _C_ is undispatched: if the limit is set to 2, and the query returns _A_ and _B_ we won't be able to figure out whether _A_ and _B_ are blocked on _C_ since we don't yet know if _C_ is blocked on _D_.

Reply:
I think it needs to be restartable, at the very least (and if its restartable, then it can be limited, I'd presume.) Our limit would probably be in the thousands or tens of thousands, and I feel like it's reasonable to have it creep a bit and do an extra query for C

Reply:
I added a limit parameter.
To solve the issue of incomplete versions I added a second query to round out the tasks with the rest of the matching tasks in their versions (in case they've been omitted by the limit).